drivers/display/ssd1306.c

```diff
@@ -48,56 +48,75 @@ LOG_MODULE_REGISTER(ssd1306, CONFIG_DISPLAY_LOG_LEVEL);
 #define SSD1306_ADDRESSING_MODE		(SSD1306_SET_MEM_ADDRESSING_HORIZONTAL)
 #endif
 
+union ssd1306_bus {
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+	struct i2c_dt_spec i2c;
+#endif
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+	struct spi_dt_spec spi;
+#endif
+};
```

Comment:
Removed #if in SPI and I2C conditionals.
However, the following parts are left because they cause unreferenced warnings.

https://github.com/zephyrproject-rtos/zephyr/pull/56887/files#diff-7da6d958478082c2b7f2549d889797958fbf4226bbee0fadbf40e3907479afadR68
https://github.com/zephyrproject-rtos/zephyr/pull/56887/files#diff-7da6d958478082c2b7f2549d889797958fbf4226bbee0fadbf40e3907479afadR94